src/trace/dma-trace.c

```diff
@@ -278,10 +278,26 @@ static int dma_trace_start(struct dma_trace_data *d)
 		return -ENODEV;
 	}
 
-	err = dma_copy_set_stream_tag(&d->dc, d->stream_tag);
+	if (d->dc.chan) {
+		/* We already have DMA channel for dtrace, stop it */
+		tr_info(&dt_tr, "dma_trace_start(): DMA channel reconfiguration");
+		err = dma_stop(d->dc.chan);
+
+		/* Re-request a channel if different tag is provided */
+		if (d->active_stream_tag != d->stream_tag) {
+			dma_channel_put(d->dc.chan);
+			d->dc.chan = NULL;
+			err = dma_copy_set_stream_tag(&d->dc, d->stream_tag);
```

Comment:
@ranj063, I think they will complement each other nicely. Do you see a conflict between the two in code wise?

The full solution will be to properly release the dtrace, this patch 'fixes' it under the hood, but it might be a good sanity protection.
The firmware/linux started to act strangely afyer the dtrace reconfig failed on my tests.